packages/legend-application-data-cube/src/stores/builder/LegendDataCubeBuilderStore.tsx

```diff
@@ -295,6 +356,15 @@ export class LegendDataCubeBuilderStore {
         const specification = DataCubeSpecification.serialization.fromJson(
           persistentDataCube.content,
         );
+
+        if (this.loader.isPartialSouce(specification.source._type as string)) {
```

Comment:
As discussed, let's make this code path declarative and not dancing around various state for the blocking window state.